crates/proto/src/op/query.rs

```diff
@@ -188,22 +188,33 @@ pub struct QueryParts {
 
 impl From<Query> for QueryParts {
     fn from(q: Query) -> Self {
-        cfg_if::cfg_if! {
-            if #[cfg(feature = "mdns")] {
-                let Query {
-                    name,
-                    query_type,
-                    query_class,
-                    mdns_unicast_response,
-                } = q;
-            } else {
-                let Query {
-                    name,
-                    query_type,
-                    query_class,
-                } = q;
-            }
+        let Query {
```

Comment:
This stuff makes me wonder if it really makes sense to have both `Query` and `QueryParts`? Maybe we should just make `Query`'s fields public (maybe stick a `#[non_exhaustive]` on it) and ditch `QueryParts`?

Reply:
Looking back at the history, #1397 introduced `QueryParts` in order to destructure a `Query` and use its buffers without copying them to new allocations. All four setter methods on `Query` don't do any validation, so the type isn't maintaining any invariants. I think making the fields public makes sense, how about I do so in a follow-up PR?